src/kernel/src/ucred/mod.rs

```diff
@@ -1,25 +1,40 @@
 pub use self::auth::*;
 pub use self::id::*;
+use self::prison::Prison;
+use self::prison::PRISON0;
 pub use self::privilege::*;
+use crate::errno::EAFNOSUPPORT;
+use crate::errno::ESRCH;
 use crate::errno::{Errno, EPERM};
+use crate::net::AddressFamily;
+use crate::ucred::prison::PrisonAllow;
+use crate::ucred::prison::PrisonFlags;
 use std::num::NonZeroI32;
 use thiserror::Error;
 
 mod auth;
 mod id;
+pub mod prison;
 mod privilege;
 
 /// An implementation of `ucred` structure.
 #[derive(Debug, Clone)]
 pub struct Ucred {
-    effective_uid: Uid, // cr_uid
-    real_uid: Uid,      // cr_ruid
-    groups: Vec<Gid>,   // cr_groups + cr_ngroups
+    effective_uid: Uid,      // cr_uid
+    real_uid: Uid,           // cr_ruid
+    groups: Vec<Gid>,        // cr_groups + cr_ngroups
+    prison: &'static Prison, // cr_prison
```

Comment:
How the prison is specified if the process is jailed? If it is allocated a `Prison` then using a static reference here is not correct.